data_model/primitives/src/lib.rs

```diff
@@ -15,3 +15,91 @@ extern crate alloc;
 pub mod atomic;
 pub mod fixed;
 pub mod small;
+
+use fixed::prelude::*;
+
+/// Trait for values that could be converted to `f64` and observed by prometheus.
+pub trait FloatMetricFrom {
+    /// Convert `value` to `f64`.
+    fn from(value: Self) -> f64;
+}
```

Comment:
I'd prefer `IntoMetric` and `fn into(self) -> f64` more. But it's just my opinion

Reply:
Yeah, i also think about this option, i will address your suggestion.